File: 백준/Bronze/2748. 피보나치 수 2/피보나치 수 2.rs
fn fib(n: u8) -> u64 {
    if n <= 1 {
        return n as u64;
    }

    let mut prev = 0;
    let mut cur = 1;

    for _ in 2..=n {
        let next = prev + cur;
        prev = cur;
        cur = next;
    }

    cur
}

fn main() {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    let n: u8 = input.trim().parse().unwrap();

    println!("{}", fib(n))
}

File: 백준/Bronze/2748. 피보나치 수 2/피보나치 수 2.py
import sys

input = sys.stdin.readline
dp = [0, 0, 1]


def fibonacci(N: int):
    global dp

    while len(dp) < N + 2:
        dp.append(dp[-1] + dp[-2])

    return dp[N + 1]


if __name__ == "__main__":
    N = int(input())
    print(fibonacci(N))
